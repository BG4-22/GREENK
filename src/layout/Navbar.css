.linkTypeOne {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    background-color: #8ba5ff;
    border-radius: 3rem;
}

.linkTypeTwo {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    height: 135%;
    margin-bottom: 25px;
    width: 30%;
    border-radius: 3em;
    border: 0.1px solid black;
    background-color: #8ba5ff;
}
