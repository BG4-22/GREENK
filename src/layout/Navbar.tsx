import { Box, Button, Center, Grid, GridItem, Text } from '@chakra-ui/react';
import { FiHome } from 'react-icons/fi';
import { Link } from 'react-router-dom';

const Navbar: React.FC = () => {
    return (
        <Box
            aria-label={'navigation'}
            display={'flex'}
            justify-content={'center'}
            alignItems={'center'}
            position={'relative'}
            bg={'#6387ff'}
            borderRadius={70}
            border={'.1px solid black'}
            w={'100%'}
            h={'3rem'}
            maxW={'50%'}
            margin={'6em auto auto auto'}
            bottom={'1rem'}
            color="white">
            <Grid h="100%" w="100%" templateColumns="1fr auto 1fr">
                <GridItem h="100%" w="100%" alignSelf="center">
                    <Button
                        h="100%"
                        bg={'#6387ff'}
                        display={'flex'}
                        flexDirection={'column'}
                        borderRadius={70}
                        w={'100%'}
                        _hover={{
                            bg: '#8BA5FF',
                            border: '.1px solid #000',
                            color: '#000',
                        }}>
                        <Link
                            to="/energiflyt"
                            style={{
                                textDecoration: 'none',
                            }}>
                            Energiflyt
                        </Link>
                    </Button>
                </GridItem>
                <GridItem h="100%" w="100%">
                    <Center w={'100px'}>
                        <Link
                            to="/"
                            style={{
                                textDecoration: 'none',
                            }}>
                            <Button
                                id={'navbar-home-button'}
                                position={'absolute'}
                                left={'50%'}
                                transform={'translateX(-50%)'}
                                bottom={'0px'}
                                h={'110px'}
                                w={'110px'}
                                zIndex={1}
                                bg={'#6387ff'}
                                minH={'100%'}
                                border={'.1px solid black'}
                                borderRadius={'1000px'}
                                colorScheme="blue"
                                aria-label="Home button"
                                _hover={{ bg: '#8BA5FF' }}>
                                <Box
                                    justifyContent={'center'}
                                    alignItems={'center'}
                                    display={'flex'}
                                    flexDirection={'column'}>
                                    <FiHome size={30} title={'Hjem'} />
                                    <Text>Hjem</Text>
                                </Box>
                            </Button>
                        </Link>
                    </Center>
                </GridItem>
                <GridItem h="100%" w="100%" alignSelf="center">
                    <Link
                        to="/statistikk"
                        style={{
                            textDecoration: 'none',
                        }}>
                        <Button
                            h="100%"
                            bg={'#6387ff'}
                            borderRadius={70}
                            w={'100%'}
                            _hover={{
                                bg: '#8BA5FF',
                                border: '.1px solid #000',
                                color: '#000',
                            }}>
                            Statistikk
                        </Button>
                    </Link>
                </GridItem>
            </Grid>
        </Box>
    );
};

export default Navbar;
