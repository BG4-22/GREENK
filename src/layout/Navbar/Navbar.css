.link {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    background-color: #8ba5ff;
    border-radius: 3rem;
}

#navBar {
    display: flex;
    justify-content: space-around;
    align-items: center;
    position: relative;
    background-color: #8ba5ff;
    border-radius: 100px;
    height: 75px;
    width: 100%;
    max-width: 50%;
    color: white;
    margin-top: 10px;
    font-size: 1.5rem;
}

#navBar,
.homeButton {
    border-bottom: 3px solid grey;
}

.homeButton {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    transform: translateY(-0.8rem);
    border-radius: 100px;
    background-color: #8ba5ff;
}
