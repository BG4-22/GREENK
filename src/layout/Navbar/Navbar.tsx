import { Box, Circle, Text } from '@chakra-ui/react';
import { FiHome } from 'react-icons/fi';
import { Link } from 'react-router-dom';
import './Navbar.css';

/**
 * Component that provides links to each url path. Decides what component that is shown.
 * @returns Navbar
 */
const Navbar: React.FC = () => {
    return (
        <Box id={'navBar'} aria-label="navigation">
            <Link className="link" to={'/energiflyt'}>
                Energiflyt
            </Link>
            <Link to={'/'}>
                <Circle size={100} className={'homeButton'}>
                    <FiHome size={30} />
                    <Text style={{ fontSize: '15px' }}>Hjem</Text>
                </Circle>
            </Link>
            <Link className="link" to={'/statistikk'}>
                Statistikk
            </Link>
        </Box>
    );
};

export default Navbar;
