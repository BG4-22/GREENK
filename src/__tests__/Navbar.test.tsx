import Navbar from '../layout/Navbar/Navbar';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Link } from 'react-router-dom';

describe('<Navbar />', () => {
    const wrapper = shallow(<Navbar />);

    it('Should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper).toHaveLength(1);
    });

    it('Should include home-button', () => {
        const home = wrapper.find('Hjem');
        expect(home).toBeDefined();
    });

    it('Should include energyflow-button', () => {
        const statistics = wrapper.find('Energiflyt');
        expect(statistics).toBeDefined();
    });

    it('Should include statistics-button', () => {
        const statistics = wrapper.find('Statistikk');
        expect(statistics).toBeDefined();
    });
});
