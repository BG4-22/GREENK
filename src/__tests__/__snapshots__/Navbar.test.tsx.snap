// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render correctly 1`] = `
<Box
  aria-label="navigation"
  id="navBar"
>
  <Link
    className="link"
    to="/energiflyt"
  >
    Energiflyt
  </Link>
  <Link
    to="/"
  >
    <Circle
      className="homeButton"
      size={100}
    >
      <FiHome
        size={30}
      />
      <Text
        style={
          Object {
            "fontSize": "15px",
          }
        }
      >
        Hjem
      </Text>
    </Circle>
  </Link>
  <Link
    className="link"
    to="/statistikk"
  >
    Statistikk
  </Link>
</Box>
`;
